pkg/configuration/configuration.go

```diff
@@ -168,6 +170,19 @@ func (c *Config) ClearCredentials() error {
 	}
 	c.BearerToken = ""
 	c.ExpiresAt = ""
+	keys := []string{"bearer", "expires_at"}
+	allKeys := viper.AllKeys()
+	for _, k := range keys {
+		if util.InSlice(k, allKeys) {
+			viper.Set(k, "")
+		}
+	}
+	if err := viper.WriteConfig(); err != nil {
+		// only return error if there is a config file to write to
+		if !errors.As(err, &viper.ConfigFileNotFoundError{}) {
+			return err
+		}
+	}
```

Comment:
why is this needed? Can you prepend a inline comment why this is needed

Reply:
You mean the file not found check? Well, here's what happens if you don't check for it:

Reply:
I just meant add a inline code comment in this function explaining why we need this logic, so our future self does not get confused about it (speaking mainly for myself here)

Reply:
Sure, I'll do that